src/main/java/org/helpberkeley/memberdata/Main.java

```diff
@@ -709,27 +709,35 @@ private static void doUpdateMemberData(
         ApiQueryResult apiQueryResult = HBParser.parseQueryResult(json);
         Map<String, DetailsPost> deliveryDetails = HBParser.deliveryDetails(apiQueryResult);
 
-        String workflowFileName = request.uploadFile.getFileName();
-        String deliveries = apiClient.downloadFile(workflowFileName);
+        String originalWorkflowFileName = request.uploadFile.getOriginalFileName();
+        String updatedFileName = originalWorkflowFileName.replace(".csv", "-upd.csv");
+        String fileName = request.uploadFile.getFileName();
+        String deliveries = apiClient.downloadFile(fileName);
         request.postStatus(WorkRequestHandler.RequestStatus.Processing, "");
 
-        WorkflowParser parser = WorkflowParser.create(Collections.emptyMap(), deliveries);
-        WorkflowExporter exporter = new WorkflowExporter(parser);
+        WorkflowParser parser;
+        WorkflowExporter exporter;
         String updatedCSVData;
         try {
+            parser = WorkflowParser.create(Collections.emptyMap(), deliveries);
+            exporter = new WorkflowExporter(parser);
             updatedCSVData = exporter.updateMemberData(users, deliveryDetails);
         } catch (MemberDataException ex) {
             LOGGER.warn("updatedMemberData failed: " + ex + "\n" + ex.getMessage());
             request.postStatus(WorkRequestHandler.RequestStatus.Failed, ex.getMessage());
             return;
         }
-
-        exporter.writeFile(workflowFileName, updatedCSVData);
-        Upload upload = new Upload(apiClient, workflowFileName);
-        String statusMessage = "Spreadsheet with updated member data uploaded: ["
-                + workflowFileName + "|attachment](" + upload.getShortURL() + ")\n"
-                + "\n" + "The table below shows which data was updated: \n" + exporter.getWarnings();
-
+        String statusMessage;
+        if (exporter.getUpdatedUsers().isEmpty()) {
+            statusMessage = "The uploaded spreadsheet " + originalWorkflowFileName + " is already up-to-date. " +
+                    "There are no changes/updates for these members.";
```

Comment:
Use a constant and MessageFormat to prevent string duplication brittleness in the test case(s).